awx/api/views/__init__.py

```diff
@@ -3943,6 +3943,10 @@ def get(self, request, **kwargs):
 
         prev_non_meta_event = events[0]
         for i, e in enumerate(events):
+            if not e['event']:
+                logging.warning(f'event type missing for event {e}')
+                continue
```

Comment:
I think my only remaining question is if this `continue` is _necessary_ or if we could just handle these events within the rest of the logic. But either way.